packages/hoppscotch-app/components/http/ImportCurl.vue

```diff
@@ -124,7 +138,35 @@ const handleImport = () => {
       }
     }
 
-    const method = parsedCurl.method.toUpperCase()
+    const method = parsedCurl.method?.toUpperCase() || "GET"
+    const contentType = parsedCurl.contentType
+    const auth: HoppRESTAuth = getAuthObject(parsedCurl, username, password)
+
+    // append danglingParams to url
+    if (danglingParams.length > 0) endpoint += "?" + danglingParams.join("&")
+
+    // final body if multipart data is not present
+    let finalBody = {
+      contentType,
+      body: body as string | null,
+    } as HoppRESTReqBody
+
+    // if multipart data is present
+    if (Object.keys(parsedCurl.multipartUploads).length > 0) {
+      const ydob: FormDataKeyValue[] = []
+      for (const key in parsedCurl.multipartUploads) {
+        ydob.push({
+          active: true,
+          isFile: false,
+          key,
+          value: parsedCurl.multipartUploads[key],
+        })
+      }
+      finalBody = {
+        contentType: "multipart/form-data",
+        body: ydob,
+      } as HoppRESTReqBodyFormData
+    }
```

Comment:
It would be cool if we don't keep all the processing steps in the UI, let the UI worry about presentation and working with the state stores.

If we can actually have a separate function which can take the input string and spit out the HoppRESTRequest object, that would be the most ideal situation